.github/actions/smart-ci/smart_ci.py

```diff
@@ -184,6 +186,13 @@ def main():
     all_defined_components = components_config.keys()
     changed_component_names = set(all_defined_components) if run_full_scope else \
         get_changed_component_names(pr, all_possible_components, args.pattern)
+
+    core_components = ['Core', 'transformations', 'preprocessing', 'inference']
```

Comment:
I prefer to explicitly list these components in `components.yml` and avoid hardcoding in processing script.